flyway.go

```diff
@@ -88,66 +99,99 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
+	}
+
+	if len(req.Files) == 0 {
+		return fmt.Errorf("missing migrations: no files provided. %s", migrationsErrMessage)
+	} else {
+		migrationsFound := false
+		for _, file := range req.Files {
+			if file.ContainerFilePath == DefaultMigrationsPath {
+				migrationsFound = true
+			}
+		}
+
+		if !migrationsFound {
+			return fmt.Errorf("missing migrations: %s", migrationsErrMessage)
+		}
+	}
+
+	// parse connection settings
+	if req.Env[flywayEnvUrlKey] == "" {
+		return fmt.Errorf("missing database url: environment variable %s is empty", flywayEnvUrlKey)
+	}
+	if req.Env[flywayEnvUserKey] == "" {
+		return fmt.Errorf("missing user: environment variable %s is empty", flywayEnvUserKey)
+	}
+	if req.Env[flywayEnvPasswordKey] == "" {
+		return fmt.Errorf("missing password: environment variable %s is empty", flywayEnvPasswordKey)
+	}
+
+	return nil
+}
+
+func WithUser(user string) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("FLYWAY_USER", user)
 }
 
-func WithEnvPassword(password string) testcontainers.CustomizeRequestOption {
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("FLYWAY_PASSWORD", password)
 }
 
-func WithEnvUrl(dbUrl string) testcontainers.CustomizeRequestOption {
+func WithDatabaseUrl(dbUrl string) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("FLYWAY_URL", dbUrl)
 }
 
-func WithEnvGroup(group string) testcontainers.CustomizeRequestOption {
+func WithTimeout(timeout time.Duration) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.WaitingFor = wait.ForAll(
+			wait.ForExit().WithExitTimeout(timeout),
+			waitForApplied,
+			waitForValidated,
+		)
+
+		return nil
+	}
+}
+
+func WithGroup(group string) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("GROUP", group)
 }
 
-func WithEnvTable(table string) testcontainers.CustomizeRequestOption {
+func WithTable(table string) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("FLYWAY_TABLE", table)
 }
 
-func WithEnvConnectRetries(retries int) testcontainers.CustomizeRequestOption {
+func WithConnectRetries(retries int) testcontainers.CustomizeRequestOption {
 	return withEnvSetting("FLYWAY_CONNECT_RETRIES", strconv.Itoa(retries))
 }
 
-func WithEnvLocations(locations string) testcontainers.CustomizeRequestOption {
-	return withEnvSetting("FLYWAY_LOCATIONS", locations)
-}
-
 func withEnvSetting(key, group string) testcontainers.CustomizeRequestOption {
-	return func(req *testcontainers.GenericContainerRequest) error {
-		req.Env[key] = group
-		return nil
-	}
+	return testcontainers.WithEnv(map[string]string{
+		key: group,
+	})
 }
 
-func WithNetwork(network string) testcontainers.CustomizeRequestOption {
+func WithMigrations(absHostFilePath string) testcontainers.CustomizeRequestOption {
```

Comment:
I think this is the one change I was unsure about, but again its not a very significant thing. I was allowing the caller of the service to define the path inside the container to the migrations, but by doing this you are fixing them to the default path.

NOTE: I'm pretty sure I hit problems previously where the local path had to be follow the same convention, so may be I need to test that too e.g. .../flyway/sql

Reply:
I'd advocate for having opinionated modules that offers a deep API to configure them. If flyway expects the migrations in one place, why adding complexity and allowing the user to change it? In the end what it's important is the user files, not the container files.